lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -322,6 +348,53 @@ hdPassphrase (Key rootXPub) = Passphrase $
     (unXPub rootXPub)
     ("address-hashing" :: ByteString)
 
+toAddress
+    :: Key 'RootK XPub
+    -> Key 'AddressK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> Address
+toAddress rootXPub (Key addrXPub) accIx addrIx =
+    Address $ encodeBase58 bitcoinAlphabet
```

Comment:
We don't manipulate encoded addresses. The payload inside `Address` is always free of any user encoding. The encoding only happens at the boundaries of the system, when addresses have to be presented to users.